crates/cext/src/transpiler/target.rs

```diff
@@ -869,6 +872,471 @@ pub unsafe extern "C" fn qk_target_num_instructions(target: *const Target) -> us
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+/// Behavior is undefined if ``inst_props`` does not point to an address of the correct size to
+/// store ``QkInstructionProperties`` in.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_op_get_props(
+    target: *const Target,
+    inst_idx: usize,
+    qarg_idx: usize,
+    inst_props: *mut CInstructionProperties,
+) -> ExitCode {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target_borrowed = unsafe { const_ptr_as_ref(target) };
+    if let Some((_, props_map)) = target_borrowed.get_by_index(inst_idx) {
+        if let Some((_, retrieved_props)) = props_map.get_index(qarg_idx) {
+            // SAFETY: Per documentation, the pointer goes to an address
+            // pre-allocated for a ``CInstructionProperties`` object.
+            unsafe {
+                if let Some(props) = retrieved_props {
+                    write(
+                        inst_props,
+                        CInstructionProperties {
+                            duration: props.duration.unwrap_or(f64::NAN),
+                            error: props.error.unwrap_or(f64::NAN),
+                        },
+                    );
+                } else {
+                    // If no property is found but there are qargs, clear the instruction to represent a null instance.
+                    qk_target_inst_props_clear(inst_props);
+                }
+            }
+            ExitCode::Success
+        } else {
+            ExitCode::IndexError
+        }
+    } else {
+        ExitCode::IndexError
+    }
+}
+
+/// A representation of a Target operation's instruction properties.
+#[repr(C)]
+pub struct CInstructionProperties {
+    duration: f64,
+    error: f64,
+}
+
+/// @ingroup QkTarget
+/// Clear the data in the ``QkInstructionProperties`` object.
+///
+/// This function does not free the allocation for the provided ``QkInstructionProperties``
+/// pointer, but it resets the internal values to ``NAN``.
+///
+/// @param inst_props A pointer to the ``QkInstructionProperties`` object to clear.
+///
+/// # Example
+/// ```c
+///     QkTarget *target = qk_target_new(5);
+///
+///     QkTargetEntry *entry = qk_target_entry_new(QkGate_CX);
+///     uint32_t qargs[2] = {0, 1};
+///     qk_target_entry_add_property(entry, qargs, 2, 0.0, 0.1);
+///     qk_target_add_instruction(target, entry);
+///
+///     QkInstructionProperties inst_props;
+///     qk_target_op_get_props(target, 0, 0, &inst_props);
+///     
+///     // free after usage
+///     qk_target_inst_props_clear(&inst_props);
+/// ```
+///
+/// # Safety
+///
+/// Behavior is undefined if ``inst_props`` is a valid, non-null pointer
+/// to ``QkInstructionProperties``.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_inst_props_clear(inst_props: *mut CInstructionProperties) {
```

Comment:
The "clear" functions are for freeing heap memory owned by the struct. Since this struct doesn't store any pointers to heap memory, we shouldn't need it at all 🙂. 

E.g. we need a "clear" function for `QkCircuitInstruction` because it holds pointers to the following things on the heap that only it owns:

https://github.com/Qiskit/qiskit/blob/7b6c220665690e95c070efc7d704fb7a9fe0a818/crates/cext/src/circuit.rs#L839-L846